### ISSUE TITLE
Feature Request: Support NFS File Shares

### ISSUE THREAD

**Issue by rhuanbarreto** [microsoft/azure-container-apps#717]
**Is your feature request related to a problem? Please describe**.  

Some container volumes rely on NFS file shares. For example in #704, it's not possible to support MongoDB databases.

**Describe the solution you'd like**.    

I'd like to have support for NFS Azure File Shares in Azure Container Apps.

**Describe alternatives you've considered**.   

Today only alternative is to go to AKS with persistent volumes.

**Comment by fabiankaimer** [comment]
Same goes for other databases as well. PostgreSQL for example requires hardlinks to work, which is currently only supported by NFS file shares. See: https://github.com/docker-library/postgres/issues/548#issuecomment-1001513872

**Comment by rhuanbarreto** [comment]
In addition, SMB shares (even in the Premium Tier) have a very poor performance for container apps. Hope this can be solved by using NFS shares.

**Comment by kahawai-sre** [comment]
Any progress on this? As above, NFS v4 is a required standard for many modern linux-based container apps needing shared/persistent storage.

**Comment by GP-CAN** [comment]
Same on our side need Azure File NFS share in an ACA but it fails to start.

**Comment by fabiankaimer** [comment]
Any chance to somehow vote this up or get some attention from Microsoft on this topic?

**Comment by cwe1ss** [comment]
The ARM reference already contains a value „NfsAzureFile“ for the field „storageType“. Is this something we can already play around with in private / public preview?

https://learn.microsoft.com/en-us/azure/templates/microsoft.app/containerapps?pivots=deployment-language-bicep#volume

**Comment by fabiankaimer** [comment]
I got confirmation today, that the feature is currently in preview:

![image](https://github.com/microsoft/azure-container-apps/assets/15916113/274eb4b6-2632-4c2d-a58f-2f307b96ab13)

Refer to: https://learn.microsoft.com/en-us/azure/container-apps/storage-mounts?tabs=nfs&pivots=azure-cli#azure-files

Thanks guys, for taking care of that :)

**Comment by anthonychu** [comment]
Yes, we added NFS to the docs this week and it's currently in preview. Please reply here or create a new issue if you run into any problems or have feedback.

**Comment by riskpoint-per** [comment]
NFS file shares has been in preview for over a month now, but still do not for cases like PostgreSQL which requires hardlinks.

**Comment by MartyWind** [comment]
@riskpoint-per @rhuanbarreto apparently it left preview last month, can you confirm that it works now? https://github.com/MicrosoftDocs/azure-docs/commit/aa26c6577ad3be349d6ce8fed1315b01304a3f44

**Comment by riskpoint-per** [comment]
@MartyWind I moved to aks, with container apps I ended up having to come up with hacks around it for most cases

**Comment by rhuanbarreto** [comment]
Same for me. I went for AKS with disks because NFS storage is too expensive and hacky.

But for the sake of the issue it can be closed as completed.

**Comment by Ily83** [comment]
Why the need to disable the secure transfer?
